### PR TITLE
Add MPNet ONNX export

### PR DESCRIPTION
**Pull request by jplu**
# What does this PR do?
This PR adds the possibility to export MPNet models into the ONNX format.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/exporters/onnx/model_configs.py [hunk 1]
```diff
@@ -111,6 +111,10 @@ def inputs(self) -> Mapping[str, Mapping[int, str]]:
         return {"input_ids": dynamic_axis, "attention_mask": dynamic_axis}
 
 
+class MPNetOnnxConfig(DistilBertOnnxConfig):
+    DEFAULT_ONNX_OPSET = 12
```
Comment (fxmarty) [inline optimum/exporters/onnx/model_configs.py:115]: Just for my knowledge, is opset 11 failing for this model? If you want, you can add a comment as to why opset 12 is required.

**Review by fxmarty — APPROVED** [review]
Thank you for your contribution, looks good to me as long as the tests pass!

File: optimum/exporters/onnx/model_configs.py (hunk 1 above)
Comment (jplu) [reply optimum/exporters/onnx/model_configs.py:115]: Here the error when export with opset 11:
```
An error occured with the error message: [ONNXRuntimeError] : 10 : INVALID_GRAPH : Load model from here/model.onnx failed:This is an invalid model. Type Error: Type 'tensor(int64)' of input parameter (/encoder/Add_1_output_0) of operator (Min) in node (/encoder/Min) is invalid..
```
Apparently the Min operator is not compliant with int64 tensors (as far as I understand this message)

**Comment by jplu** [comment]
Looks like there an issue with the multiple_choice task:
```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.15/x64/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/hostedtoolcache/Python/3.8.15/x64/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/opt/hostedtoolcache/Python/3.8.15/x64/lib/python3.8/site-packages/optimum/exporters/onnx/__main__.py", line 194, in <module>
    main()
  File "/opt/hostedtoolcache/Python/3.8.15/x64/lib/python3.8/site-packages/optimum/exporters/onnx/__main__.py", line 74, in main
    model = TasksManager.get_model_from_task(task, args.model, framework=args.framework, cache_dir=args.cache_dir)
  File "/opt/hostedtoolcache/Python/3.8.15/x64/lib/python3.8/site-packages/optimum/exporters/tasks.py", line 963, in get_model_from_task
    model = model_class.from_pretrained(model_name_or_path, **kwargs)
  File "/opt/hostedtoolcache/Python/3.8.15/x64/lib/python3.8/site-packages/transformers/models/auto/auto_factory.py", line 463, in from_pretrained
    return model_class.from_pretrained(
  File "/opt/hostedtoolcache/Python/3.8.15/x64/lib/python3.8/site-packages/transformers/modeling_utils.py", line 2379, in from_pretrained
    ) = cls._load_pretrained_model(
  File "/opt/hostedtoolcache/Python/3.8.15/x64/lib/python3.8/site-packages/transformers/modeling_utils.py", line 2695, in _load_pretrained_model
    raise RuntimeError(f"Error(s) in loading state_dict for {model.__class__.__name__}:\n\t{error_msg}")
RuntimeError: Error(s) in loading state_dict for MPNetForMultipleChoice:
	size mismatch for classifier.weight: copying a param with shape torch.Size([2, 64]) from checkpoint, the shape in current model is torch.Size([1, 64]).
	size mismatch for classifier.bias: copying a param with shape torch.Size([2]) from checkpoint, the shape in current model is torch.Size([1]).
	You may consider adding `ignore_mismatched_sizes=True` in the model `from_pretrained` method.
```

@fxmarty any idea how to fix this? Or should I remove this tasks from the task manager?

**Comment by fxmarty** [comment]
@jplu It's probably just an issue with `hf-internal-testing/tiny-random-mpnet`. Could you try using https://huggingface.co/hf-internal-testing/tiny-random-MPNetModel ?

You should be able to run the failing test specifying `-k "test_exporters_cli_pytorch_273_mpnet_multiple_choice"` in pytest

**Comment by jplu** [comment]
The test is skiped:
```
pytest tests/exporters/test_exporters_onnx_cli.py -k "test_exporters_cli_pytorch_273_mpnet_multiple_choice"
=================================================================================================================================================================================== test session starts ====================================================================================================================================================================================
platform linux -- Python 3.10.8, pytest-7.2.0, pluggy-1.0.0
rootdir: /home/jplu/dev/perso/optimum, configfile: setup.cfg
collected 413 items / 412 deselected / 1 selected

tests/exporters/test_exporters_onnx_cli.py s                                                                                                                                                                                                                                                                                                                                         [100%]

============================================================================================================================================================================ 1 skipped, 412 deselected in 1.41s ============================================================================================================================================================================
```

**Comment by fxmarty** [comment]
Hum that's weird. What if you just pass `-k "mpnet"`?

**Comment by jplu** [comment]
They are all skipped:
```
=================================================================================================================================================================================== test session starts ====================================================================================================================================================================================
platform linux -- Python 3.10.8, pytest-7.2.0, pluggy-1.0.0
rootdir: /home/jplu/dev/perso/optimum, configfile: setup.cfg
collected 413 items / 406 deselected / 7 selected

tests/exporters/test_exporters_onnx_cli.py sssssss                                                                                                                                                                                                                                                                                                                                   [100%]

============================================================================================================================================================================ 7 skipped, 406 deselected in 2.62s ============================================================================================================================================================================
```

**Comment by fxmarty** [comment]
I can run the mpnet tests locally on your branch with `pytest tests/exporters/test_exporters_onnx_cli.py -k "mpnet" --exitfirst -s`. Not sure what's wrong

**Comment by fxmarty** [comment]
It's all good, thanks for the addition!

**Pull request merged**